From: Michal Kubecek <mkubecek@suse.cz>
Date: Tue, 12 May 2020 22:10:19 +0200
Subject: netlink: fix build warnings
Patch-mainline: Not yet, going to submit for 5.7 cycle

Depending on compiler version and options, some of these warnings may
result in build failure.

- gcc 4.8 wants __KERNEL_DIV_ROUND_UP defined before including ethtool.h
- avoid pointer arithmetic on void *
- do not use printf(s) if string is not a string literal

Signed-off-by: Michal Kubecek <mkubecek@suse.cz>
---
 netlink/desc-ethtool.c | 2 +-
 netlink/parser.c       | 2 +-
 netlink/settings.c     | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

--- a/netlink/desc-ethtool.c
+++ b/netlink/desc-ethtool.c
@@ -4,9 +4,9 @@
  * Descriptions of ethtool netlink messages and attributes for pretty print.
  */
 
+#include "../internal.h"
 #include <linux/ethtool_netlink.h>
 
-#include "../internal.h"
 #include "prettymsg.h"
 
 static const struct pretty_nla_desc __header_desc[] = {
--- a/netlink/parser.c
+++ b/netlink/parser.c
@@ -1016,7 +1016,7 @@ int nl_parser(struct nl_context *nlctx, const struct param_parser *params,
 			buff = tmp_buff_find(buffs, parser->group);
 		msgbuff = buff ? &buff->msgbuff : &nlsk->msgbuff;
 
-		param_dest = dest ? (dest + parser->dest_offset) : NULL;
+		param_dest = dest ? ((char *)dest + parser->dest_offset) : NULL;
 		ret = parser->handler(nlctx, parser->type, parser->handler_data,
 				      msgbuff, param_dest);
 		if (ret < 0)
--- a/netlink/settings.c
+++ b/netlink/settings.c
@@ -375,7 +375,7 @@ static int dump_link_modes(struct nl_context *nlctx,
 after:
 	if (first && if_none)
 		printf("%s", if_none);
-	printf(after);
+	fputs(after, stdout);
 
 	return 0;
 err:
